File: commonutils/src/main/java/net/maxsmr/commonutils/android/preferences/PreferenceChangeListener.java
package net.maxsmr.commonutils.android.preferences;


import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public interface PreferenceChangeListener {
    <T> void onPreferenceChanged(String preferenceName, @NotNull String key, @Nullable T oldValue, @Nullable T newValue);

    void onPreferenceRemoved(String preferenceName, @NotNull String key);

    void onAllPreferencesRemoved(String preferenceName);
}

File: commonutils/src/main/kotlin/net/maxsmr/commonutils/prefs/PreferenceChangeListener.kt
package net.maxsmr.commonutils.prefs

interface PreferenceChangeListener {

    fun <T> onPreferenceChanged(name: String, key: String, oldValue: T?, newValue: T?, prefType: SharedPrefsHolder.PrefType?)

    fun onPreferenceRemoved(name: String, key: String)

    fun onAllPreferencesRemoved(name: String)

    fun onAllPreferencesRefreshed(name: String)
}
